Examples/CMakeLists.txt

```diff
@@ -12,6 +12,12 @@ if(WarpX_MPI)
         if(${MPIEXEC_HELP_TEXT} MATCHES "^.*allow-run-as-root.*$")
             set(MPI_ALLOW_ROOT --allow-run-as-root)
         endif()
+        if(${MPIEXEC_HELP_TEXT} MATCHES "^.*--host.*$")
+            # Avoid the following error seen with Open MPI 4.1.6 in the Azure pipelines:
+            # "There are not enough slots available in the system to satisfy the 2 slots
+            # that were requested by the application"
+            set(MPI_HOST_NUMPROC --host localhost:${nprocs})
+        endif()
```

Comment:
@ax3l @WeiqunZhang @atmyers 

Do you know any other way to check whether a given MPI executable supports a given option, `--host` in this case, to avoid passing it when it is not supported?

The solution I implemented here does not seem to work in this case, although we use it for other options (see code right above).